pkg/didcomm/messenger/messenger.go

```diff
@@ -195,19 +198,23 @@ func (m *Messenger) ReplyTo(msgID string, msg service.DIDCommMsgMap) error {
 // ReplyToNested sends the message by starting a new thread.
 // Do not provide a message with ~thread decorator. It will be rewritten.
 // The function adds ~thread decorator to the message according to the given threadID.
-// NOTE: Given threadID becomes parent threadID.
-func (m *Messenger) ReplyToNested(threadID string, msg service.DIDCommMsgMap, myDID, theirDID string) error {
+// NOTE: Given threadID (from opts or from message record) becomes parent threadID.
+func (m *Messenger) ReplyToNested(msg service.DIDCommMsgMap, opts *service.NestedReplyOpts) error {
```

Comment:
I would suggest instead of using `*service.NestedReplyOpts` we need to use variadic args.
`NestedReplyOpts` should be private. 
Two functions need to be exposed.
e.g `WithMsgID(MsgID)` and with `WithDID(TheirDID,MyDID,ThreadID)`

It will force the client to use the right option without looking into the implementation.
Otherwise, It will confuse what data needs to be populated.
Someone might think about to provide custom `ThreadID` and `MsgID`. e.g
``` 
&NestedReplyOpts {
	ThreadID: "...."
	MsgID:"...."
}
```
The behavior here might surprise :)

Reply:
My point is to create an option function for each use case.

Reply:
I created one, but later reverted back while pushing the code,
 Since it is an interface, we have to export opts struct & resolve function as well so that implementing types can refer them from implementation.

Still, I get your point I can update it. 
In above approach we can also mix and match options.
For example: in this implementation 'TheirDID,MyDID,ThreadID'. User can pass `MsgID` & `MyDID,TheirDID` options, missing `ThreadID` will be pre-filled by record read by MsgID.